main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/aws/aws-sdk-go/service/ssm"
+)
+
+func main() {
+	lambda.Start(HandleLambdaEvent)
+}
+
+func HandleLambdaEvent() {
+	var r RealRunner
+	runReport(r)
+}
+
+func runReport(r Runner) {
+	log.SetFlags(0)
+
+	session := session.Must(session.NewSession())
+
+	if err := r.Setup(session); err != nil {
+		log.Printf("Setup error: %v", err)
+		return
+	}
+
+	dryRun := isDryRun()
+	if err := r.Run(dryRun); err != nil {
+		log.Printf("Run error: %v", err)
+		return
+	}
+
+	if !dryRun {
+		if err := r.Store(session); err != nil {
+			log.Printf("Store error: %v", err)
+			return
+		}
+	}
+}
+
+func isDryRun() bool {
+	dryRun, err := strconv.ParseBool(os.Getenv("GHTOOL_DRY_RUN"))
```

Comment:
Maybe we can move this logic from here to the CLI and having that one accepting `GHTOOL_DRY_RUN` so we do not have to have this logic here at all.